dot/parachain/overseer/overseer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package overseer
+
+import "fmt"
+
+type Overseer struct {
+	subsystems        map[Subsystem]*Context
+	subsystemMessages map[Subsystem]chan any
+}
+
+type ExamlpeSender struct {
+	senderChan chan any
+}
+
+func (s *ExamlpeSender) SendMessage(msg any) error {
+	s.senderChan <- msg
+	return nil
+}
+
+func NewOverseer() *Overseer {
+	return &Overseer{
+		subsystems:        make(map[Subsystem]*Context),
+		subsystemMessages: make(map[Subsystem]chan any),
+	}
+}
+
+func (o *Overseer) RegisterSubsystem(subsystem Subsystem) {
+	// TODO: determine best buffer size
+	comChan := make(chan any, 5)
+	o.subsystemMessages[subsystem] = comChan
+	receiverChan := make(chan any, 5)
+	o.subsystems[subsystem] = &Context{
+		Sender:   &ExamlpeSender{senderChan: comChan},
+		Receiver: receiverChan,
+	}
```

Comment:
I think we would be able to just use a common channel on which all subsystems send and overseer receives. let's call this channel `overseerChan`. We can add this as a field in parachain service struct.

```
    // Declare a channel that can only receive values of type string
    var ch1 <-chan string
    // Declare a channel that can only send values of type string
    var ch2 chan<- string
```
Let's use directional channels to make sure that we can only receive on this channel from overseer and only send from other subsystems. Let's return this `overseerChan` in `NewOverseer`.

For subsystems to receive, we would need different channels for each subsystems. I feel like these channels should be created by different subsystems and passed to overseer on registerSubsystem.

Reply:
I've updated this to share the overseerChan with subsystems, so the overseer can receive messages from subsystems.  I've added tests to simulate message passing, I think this is working the way that we intend, but I'm not sure I have used directional channels the way we envision.  Let me know your thoughts.